classes/Webp/Picture/Display.php

```diff
@@ -147,7 +156,51 @@ public function process_content( $content ) {
 			$content = str_replace( $image['tag'], $tag, $content );
 		}
 
-		return $content;
+		return $this->reinsert_picture_tags( $content );
+	}
+
+	/**
+	 * Remove pre-existing <picture> tags.
+	 *
+	 * We shouldn't replace images already nested inside picture tags
+	 * that are already in the page.
+	 *
+	 * @since 1.10.0
+	 *
+	 * @param string $html Content of the page.
+	 *
+	 * @return string HTML content without pre-existing <picture> tags.
+	 */
+	private function remove_picture_tags( $html ) {
+		return preg_replace_callback(
+			'#<picture[^>]*>.*?<\/picture\s*>#mis',
+			function( $matches ) {
+				$this->picture_tags[] = $matches[0];
+				return '<!--imagify-picture-tag-placeholder-->';
+			},
+			$html
+		);
+	}
+
+	/**
+	 * Re-insert pre-existing <picture> tags back to their original positions.
+	 *
+	 * Used in conjunction with remove_picture_tags() to re-construct HTML after having replaced non-nested images.
+	 *
+	 * @since 1.10.0
+	 *
+	 * @param string $html Content of the page without pre-existing <picture> tags.
+	 *
+	 * @return string HTML content with pre-existing <picture> tags re-inserted.
+	 */
+	private function reinsert_picture_tags( $html ) {
+		return preg_replace_callback(
```

Comment:
I'd suggest adding a bailout condition here to return early the `$html` if `$this->picture_tags` is empty

Reply:
Great Idea!